nix/default.nix

```diff
@@ -19,55 +19,29 @@ assert cudaSupport -> (cudaMajorVersion == "9" || cudaMajorVersion == "10" || cu
 let
   sources = import ./sources.nix { inherit pkgs; }
     // sourcesOverride;
-  iohKNix = import sources.iohk-nix {};
-  haskellNix = (import sources.haskell-nix { inherit system sourcesOverride; }).nixpkgsArgs;
-  # use our own nixpkgs if it exist in our sources,
-  # otherwise use iohkNix default nixpkgs.
-  nixpkgs = sources.nixpkgs-staging-next or
-    (builtins.trace "Using IOHK default nixpkgs" iohKNix.nixpkgs);
+  iohkNix = import sources.iohk-nix {};
+  haskellNix = import sources.haskell-nix { inherit system sourcesOverride; };
+
+  # Use haskell.nix default nixpkgs
+  nixpkgsSrc = haskellNix.sources.nixpkgs-2009;
```

Comment:
```suggestion
  nixpkgsSrc = haskellNix.sources.nixpkgs-unstable
```
cf. https://github.com/hasktorch/hasktorch/blob/master/nix/default.nix#L26.
This should help with the build.

Reply:
Thx! I'll try it.

Reply:
https://github.com/junjihashimoto/hasktorch-skeleton/actions
There is no reduction for nix-build. In the stack's case, it is 1 hour shorter.

![image](https://user-images.githubusercontent.com/2469618/113245307-a5b7ec80-92f1-11eb-8d56-65aa82cfb1f6.png)

Reply:
@junjihashimoto I hope that b00d914 will help; this brings the dependency versions into sync with the hasktorch repo so we should get more cache hits.

Reply:
Also, I had neglected to update the hasktorch dependency and GHC version in `stack.yaml`. Fixed that in 8c0a35e and a12460e.